gst/tensor_filter/tensor_filter.c

```diff
@@ -794,144 +721,174 @@ gst_tensor_filter_transform_ip (GstBaseTransform * trans, GstBuffer * buf)
   /** @todo 1. Resize buf if output is larger than input */
   /** @todo 2. Call the filter-subplugin callback, "invoke" */
   /** @todo 3. Return result! */
-  g_assert (1 == 0);
+  g_assert (0);
   return GST_FLOW_ERROR;
 }
 
-
 /**
- * @brief process property values, call get/set I/O dim. (internal static function)
- * If set-prop configured dimension, verify the dimension with fw callbacks
- * Otherwise, configure dimension with fw callbacks.
- *
- * @param filter "this" pointer
- * @param fixate TRUE if we may fixate property values.
- * @return 1: OK and all set. 0: Try again later. -1: cannot proceed. fatal ERROR.
+ * @brief Configure input and output tensor info.
+ * @param self "this" pointer
+ * @param incaps received caps for sink pad
+ * @return TRUE if fully configured
  */
-static int
-gst_tensor_filter_property_process (GstTensor_Filter * filter, gboolean fixate)
+static gboolean
+gst_tensor_filter_configure_tensor (GstTensorFilter * self,
+    const GstCaps * incaps)
 {
-  GstTensor_Filter_Framework *fw = filter->prop.fw;
-  GstTensor_Filter_Properties *prop = &filter->prop;
-  int ret;
-  GstTensor_TensorsMeta meta;
-  int i, tensor_idx;
-
-  /* Ensure the subplugin is contacted first before checking the XOR assert */
-  if (!prop->fwOpened && fw->open)
-    fw->open (filter, &filter->privateData);
-  prop->fwOpened = TRUE;
-
-  if (fw->getInputDimension != NULL) {
-    gst_tensor_filter_call (filter, ret, getInputDimension, &meta);
-    if (ret == 0) {
-      for (tensor_idx = 0; tensor_idx < meta.num_tensors; tensor_idx++) {
-        if (prop->inputConfigured & _TFC_TYPE)
-          if (prop->inputMeta.types[tensor_idx] != meta.types[tensor_idx]) {
-            return -1;
-          }
-        if (prop->inputConfigured & _TFC_DIMENSION)
-          for (i = 0; i < NNS_TENSOR_RANK_LIMIT; i++)
-            if (prop->inputMeta.dims[tensor_idx][i] != meta.dims[tensor_idx][i]) {
-              return -1;
-            }
-        if (fixate && !(prop->inputConfigured & _TFC_TYPE)) {
-          prop->inputMeta.types[tensor_idx] = meta.types[tensor_idx];
-          prop->inputConfigured |= _TFC_TYPE;
-        }
-        if (fixate && !(prop->inputConfigured & _TFC_DIMENSION)) {
-          memcpy (prop->inputMeta.dims[tensor_idx], meta.dims[tensor_idx],
-              sizeof (meta.dims[tensor_idx]));
-          prop->inputConfigured |= _TFC_DIMENSION;
+  GstTensorFilterProperties *prop;
+
+  prop = &self->prop;
+
+  /**
+   * GstTensorFilter has to parse the tensor dimension and type from NN model.
+   * 1. Call functions getInputDimension and getOutputDimension to get the dimension and type.
+   * 2. If these functions are not defined, call setInputDimension with parsed info from caps.
+   * 3. If set-prop configured dimension, verify the dimension with fw callbacks.
+   */
+  if (!prop->input_configured) {
```

Comment:
Because input_configured is "set" with transform_caps, you should not rely on it. The actual dimensions from other plugins may change afterwards. I'd recommend to NOT rely on input/output_configured entirely except for debugging/tracing purposes. You may rely on it ONLY if input/output_configured is set TRUE with "set_caps".

Reply:
When called transform_caps() or fixate_caps(), set the param 'incaps' NULL.